specification/monitor/resource-manager/Microsoft.Insights/preview/2024-10-01-preview/actionGroups_API.json

```diff
@@ -0,0 +1,1322 @@
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ActionDetail"
+          },
+          "description": "The list of action detail"
+        }
+      },
+      "required": [
+        "state"
+      ]
+    },
+    "ActionDetail": {
+      "description": "The action detail",
+      "type": "object",
+      "properties": {
+        "MechanismType": {
+          "type": "string",
+          "description": "The mechanism type"
+        },
+        "Name": {
+          "type": "string",
+          "description": "The name of the action"
+        },
+        "Status": {
+          "type": "string",
+          "description": "The status of the action"
+        },
+        "SubState": {
+          "type": "string",
+          "description": "The substatus of the action"
+        },
+        "SendTime": {
+          "type": "string",
+          "description": "The send time"
+        },
+        "Detail": {
+          "type": "string",
+          "description": "The detail of the friendly error message"
+        }
+      }
+    },
+    "Context": {
+      "description": "The context info",
+      "type": "object",
+      "properties": {
+        "notificationSource": {
+          "type": "string",
+          "description": "The source of the notification request"
+        },
+        "contextType": {
+          "type": "string",
+          "description": "The context id type"
+        }
+      }
+    },
+    "ActionGroupResource": {
+      "description": "An action group resource.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/AzureResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/ActionGroup",
+          "description": "The action groups properties of the resource."
+        }
+      }
+    },
+    "ActionGroupList": {
+      "description": "A list of action groups.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "x-ms-identifiers": [],
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ActionGroupResource"
+          },
+          "description": "The list of action groups."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "Provides the link to retrieve the next set of elements."
+        }
+      }
+    },
+    "NotificationRequestBody": {
+      "description": "The request body which contain contact detail metadata",
+      "type": "object",
+      "properties": {
+        "alertType": {
+          "type": "string",
+          "maxLength": 30,
+          "description": "The value of the supported alert type. Supported alert type values are: servicehealth, metricstaticthreshold, metricsdynamicthreshold, logalertv2, smartalert, webtestalert, logalertv1numresult, logalertv1metricmeasurement, resourcehealth, activitylog, actualcostbudget, forecastedbudget"
+        },
```

Comment:
If there is a fixed list of supported values - have you considered modelling this as enum?

Reply:
I don't think so. That is going to be a breaking changes. Furthermore, this is not introduced by my current changes.